tutorials/27_First_RAG_Pipeline.ipynb

```diff
@@ -0,0 +1,2555 @@
+    "\n",
+    "\n",
+    "Generators are the components that interacts with large language models (LLMs). Now, initialize a [GPTGenerator](https://docs.haystack.deepset.ai/v2.0/docs/gptgenerator) that can communicate with OpenAI GPT models with an OpenAI `api_key` and the model name you'd like to use. If you don't provide any `model_name`, the GPTGenerator defaults to `gpt-3.5-turbo`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "SavE_FAqfApo",
+    "outputId": "4c1fa657-8ca0-42ca-8e49-161edf3be505"
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from getpass import getpass\n",
+    "\n",
+    "from haystack.components.generators import GPTGenerator\n",
+    "\n",
+    "openai_api_key = os.getenv(\"OPENAI_API_KEY\", getpass(\"Enter OpenAI API key:\"))\n",
+    "generator = GPTGenerator(api_key=openai_api_key)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "nenbo2SvycHd"
+   },
+   "source": [
+    "> You can replace `GPTGenerator` in your pipeline with another `Generator`. Check out the full list of generators [here](https://docs.haystack.deepset.ai/v2.0/docs/generators)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "1bfHwOQwycHe"
+   },
+   "source": [
+    "## Building the Pipeline\n",
+    "\n",
+    "To build a pipeline, add all components to your pipeline and connect them. Create connections from `retriever` to the `prompt_builder` and from `prompt_builder` to `llm`. Explicitly connect the output of `retriever` with \"documents\" input of the `prompt_builder` to make the connection obvious as `prompt_builder` has two inputs (\"documents\" and \"question\"). For more information on pipelines and creating connections, refer to [Creating Pipelines](https://docs.haystack.deepset.ai/v2.0/docs/creating-pipelines) documentation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "f6NFmpjEO-qb"
+   },
+   "outputs": [],
+   "source": [
+    "from haystack.pipeline import Pipeline\n",
+    "\n",
+    "basic_rag_pipeline = Pipeline()\n",
+    "# Add components to your pipeline\n",
+    "basic_rag_pipeline.add_component(\"retriever\", retriever)\n",
+    "basic_rag_pipeline.add_component(\"prompt_builder\", prompt_builder)\n",
+    "basic_rag_pipeline.add_component(\"llm\", generator)\n",
+    "\n",
+    "# Now, connect the components to each other\n",
+    "basic_rag_pipeline.connect(\"retriever\", \"prompt_builder.documents\")\n",
+    "basic_rag_pipeline.connect(\"prompt_builder\", \"llm\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "mtcrMpjdzBYo"
+   },
+   "source": [
+    "### Visualize the Pipeline\n",
+    "\n",
+    "Draw the pipeline with the `draw()` method to confirm the connections are correct. You can find the diagram in the Files section of this Colab."
```

Comment:
Thank you for specifying where you can find the file! That's very helpful for newbies.